### PR TITLE
Adding initial logging infrastructure.

### PR DESCRIPTION
**Pull request by tushargupta51**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23issuecomment-84170541%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882342%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882543%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882558%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23issuecomment-84170541%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hi%20__%40tushargupta51__%2C%20I%27m%20your%20friendly%20neighborhood%20Microsoft%20Open%20Technologies%2C%20Inc.%20Pull%20Request%20Bot%20%28You%20can%20call%20me%20MSOTBOT%29.%20Thanks%20for%20your%20contribution%21%5Cr%5Cn%20%20%20%20%3Cspan%3E%5Cr%5Cn%20%20%20%20%20%20%20%20In%20order%20for%20us%20to%20evaluate%20and%20accept%20your%20PR%2C%20we%20ask%20that%20you%20sign%20a%20contribution%20license%20agreement.%20It%27s%20all%20electronic%20and%20will%20take%20just%20minutes.%20I%20promise%20there%27s%20no%20faxing.%20https%3A//cla2.msopentech.com.%5Cr%5Cn%20%20%20%20%3C/span%3E%5Cr%5Cn%5Cr%5CnTTYL%2C%20MSOTBOT%3B%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-03-20T22%3A33%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9519461%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/msotclas%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%208a998152fa80698924ae5d4f3bf369cb41e194d5%20src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882558%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Might%20be%20too%20aggressive%2C%20maybe%20Error%20as%20default%3F%22%2C%20%22created_at%22%3A%20%222015-03-20T22%3A47%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs%3AL1-76%22%7D%2C%20%22Pull%208a998152fa80698924ae5d4f3bf369cb41e194d5%20src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs%2018%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882342%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22nit%3A%20formatting%22%2C%20%22created_at%22%3A%20%222015-03-20T22%3A43%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs%3AL62-73%22%7D%2C%20%22Pull%208a998152fa80698924ae5d4f3bf369cb41e194d5%20src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105%23discussion_r26882543%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Might%20be%20a%20good%20idea%20to%20have%3A%20public%20const%20static%20EventLevel%20DefaultEventLevel%20%3D%20EventLevel.Error%22%2C%20%22created_at%22%3A%20%222015-03-20T22%3A47%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3172421%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/brentschmaltz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs%3AL1-76%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105#issuecomment-84170541'>General Comment</a></b>
- <a href='https://github.com/msotclas'></a> Hi __@tushargupta51__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
<span>
In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
</span>
TTYL, MSOTBOT;
- [ ] <a href='#crh-comment-Pull 8a998152fa80698924ae5d4f3bf369cb41e194d5 src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs 18'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105#discussion_r26882342'>File: src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs:L62-73</a></b>
- <a href='https://github.com/brentschmaltz'></a> nit: formatting
- [ ] <a href='#crh-comment-Pull 8a998152fa80698924ae5d4f3bf369cb41e194d5 src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs 12'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105#discussion_r26882543'>File: src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs:L1-76</a></b>
- <a href='https://github.com/brentschmaltz'></a> Might be a good idea to have: public const static EventLevel DefaultEventLevel = EventLevel.Error
- [ ] <a href='#crh-comment-Pull 8a998152fa80698924ae5d4f3bf369cb41e194d5 src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs 16'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105#discussion_r26882558'>File: src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs:L1-76</a></b>
- <a href='https://github.com/brentschmaltz'></a> Might be too aggressive, maybe Error as default?

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/105'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by msotclas** [comment]
Hi __@tushargupta51__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

File: src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs [hunk 1]
```diff
@@ -61,10 +62,11 @@ public class JwtSecurityTokenHandler : SecurityTokenHandler, ISecurityTokenValid
         private Int32 _maximumTokenSizeInBytes = TokenValidationParameters.DefaultMaximumTokenSizeInBytes;
         private Int32 _defaultTokenLifetimeInMinutes = DefaultTokenLifetimeInMinutes;
 
-        /// <summary>
-        /// Default lifetime of tokens created. When creating tokens, if 'expires' and 'notbefore' are both null, then a default will be set to: expires = DateTime.UtcNow, notbefore = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes).
-        /// </summary>
-        public static readonly Int32 DefaultTokenLifetimeInMinutes = 60;
+
+		/// <summary>
+		/// Default lifetime of tokens created. When creating tokens, if 'expires' and 'notbefore' are both null, then a default will be set to: expires = DateTime.UtcNow, notbefore = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes).
```
Comment (brentschmaltz) [inline src/System.IdentityModel.Tokens/Jwt/JwtSecurityTokenHandler.cs]: nit: formatting

File: src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Event source based logger to log different events.
+    /// </summary>
+    public class WilsonEventSource : EventSource
+    {
+        private static EventLevel _logLevel;
+
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs]: Might be a good idea to have: public const static EventLevel DefaultEventLevel = EventLevel.Error

File: src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        static WilsonEventSource()
+        {
+            Logger = new WilsonEventSource();
+            _logLevel = EventLevel.Informational;
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Logging/WilsonEventSource.cs]: Might be too aggressive, maybe Error as default?

**Pull request merged**